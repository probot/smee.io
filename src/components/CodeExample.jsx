import React from 'react'

const code = `<span class="hljs-keyword">const</span> SmeeClient = <span class="hljs-built_in">require</span>(<span class="hljs-string">'smee-client'</span>)

<span class="hljs-keyword">const</span> smee = <span class="hljs-keyword">new</span> SmeeClient({
<span class="hljs-attr">  source</span>: <span class="hljs-string">'${window.location.href}'</span>,
<span class="hljs-attr">  target</span>: <span class="hljs-string">'http://localhost:3000/events'</span>,
<span class="hljs-attr">  logger</span>: <span class="hljs-built_in">console</span>
})

<span class="hljs-keyword">const</span> events = smee.start()

<span class="hljs-comment">// Stop forwarding events</span>
events.close()`

export default function CodeExample () {
  return (
    <pre dangerouslySetInnerHTML={{ __html: code }} />
  )
}
