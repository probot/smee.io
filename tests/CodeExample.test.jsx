import React from 'react'
import CodeExample from '../src/components/CodeExample'
import { shallow } from 'enzyme'

describe('<CodeExample />', () => {
  describe('render', () => {
    it('renders the expected HTML', () => {
      const wrapper = shallow(<CodeExample />)
      expect(wrapper.render().text()).toMatchSnapshot()
    })
  })
})
