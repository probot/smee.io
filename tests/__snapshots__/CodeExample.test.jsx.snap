// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeExample /> render renders the expected HTML 1`] = `
"const SmeeClient = require('smee-client')

const smee = new SmeeClient({
  source: 'https://smee.io/CHANNEL',
  target: 'http://localhost:3000/events',
  logger: console
})

const events = smee.start()

// Stop forwarding events
events.close()"
`;
